lc-hax/Scripts/Patches/EnemyPatch.cs

```diff
@@ -1,10 +1,33 @@
 #pragma warning disable IDE1006
 
+using GameNetcodeStuff;
 using HarmonyLib;
 
 namespace Hax;
 
-[HarmonyPatch(typeof(EnemyAI), nameof(EnemyAI.SyncPositionToClients))]
+[HarmonyPatch(typeof(EnemyAI))]
 class EnemyPatch {
+    [HarmonyPatch(nameof(EnemyAI.SyncPositionToClients))]
     static void Prefix(ref float ___updatePositionThreshold) => ___updatePositionThreshold = 0.0f;
+
+    // Fixes game bug where you can't aim anymore if an enemy dies while attempting to kill you.
+    [HarmonyPatch(nameof(EnemyAI.CancelSpecialAnimationWithPlayer))]
+    static void Prefix(ref PlayerControllerB ___inSpecialAnimationWithPlayer) {
+        if (Helper.LocalPlayer == ___inSpecialAnimationWithPlayer) {
+            Helper.LocalPlayer.disableLookInput = false;
+        }
+    }
+}
+
+[HarmonyPatch(typeof(MaskedPlayerEnemy))]
+class MaskedPlayerEnemyPatch {
+    // Getting infected spawns a masked AI even during god mode, leading to mass cloning.
+    // To prevent this, the original is despawned, leaving only the new clone alive.
+    [HarmonyPatch(nameof(MaskedPlayerEnemy.FinishKillAnimation))]
+    static void Prefix(MaskedPlayerEnemy __instance, ref bool killedPlayer) {
+        if (__instance.inSpecialAnimationWithPlayer == Helper.LocalPlayer && killedPlayer && Setting.EnableGodMode) {
+            __instance.KillEnemyServerRpc(true);
+            __instance.HitEnemy(1000); // fallback if KillEnemy fails
```

Comment:
This line is unnecessary. `KillEnemyServerRpc` will always work. I used `HitEnemy` in `/kill` because I wanted to try getting the `Nutcracker` to drop its shotgun.

```cs
 __instance.HitEnemy(1000); // fallback if KillEnemy fails
```

Reply:
Thinking about it again, this patch would kill the Masked enemy who had tried to kill you.. Not sure if this is something we'd want.

Reply:
if a god-mode player gets caught by a masked AI, they start cloning until the player teleports away or stuns/kills them all, which is absolutely ridiculous, so there are 3 possibilities to prevent this, from easiest to hardest:
- kill the original and let the clone take its place
- stun the AI to cancel its vomit, hopefully quick enough to prevent the clone from spawning
- kill the new clone

Reply:
Why not just not make the player grabbable if they are in God mode?

Reply:
don't think that possible, the AI is controlled by the server and it only checks for a collision with an alive player to start the infection procedure, which clients can only interrupt by stunning or killing the AI

Reply:
There's a method called `IsPlayerTargettable` that we can patch.

Reply:
but `PlayerIsTargetable` is called server-side by the AI, how is that supposed to work?

Reply:
Okay, so in this game, AI is not server-sided. When an enemy targets you, it gives you ownership of the AI. This saves the server from processing the pathfinding to you. Instead, you as the client, compute the pathfinding of the enemy targeting you.

This means that when the enemy is about to target you, you retrieve ownership of the enemy and tell the enemy that you are untargetable.

Reply:
You can see this https://github.com/winstxnhdw/lc-hax/blob/main/lc-hax/Scripts/Patches/UntargetablePatch.cs

Reply:
You know what, how about I just add a check for God mode in the Untargetable Patch instead.

Reply:
interesting, so we could simply patch away the clone creation then